src/hb-ot-layout.cc

```diff
@@ -1137,3 +1159,239 @@ hb_ot_layout_substitute_lookup (OT::hb_apply_context_t *c,
 {
   apply_string<GSUBProxy> (c, lookup, accel);
 }
+
+/*
+ * OT::MATH
+ */
+
+/**
+ * hb_ot_layout_has_math_data:
+ *
+ * @face: #hb_face_t to test
+ *
+ * This function allows to verify the presence of an OpenType MATH table on the
+ * face. If so, such a table will be loaded into memory and sanitized. You can
+ * then safely call other functions for math layout and shaping.
+ *
+ * Return value: #TRUE if face has a MATH table and #FALSE otherwise
+ *
+ * Since: ????
+ **/
+hb_bool_t
+hb_ot_layout_has_math_data (hb_face_t *face)
+{
+  return &_get_math (face) != &OT::Null(OT::MATH);
+}
+
+/**
+ * hb_ot_layout_get_math_constant:
+ *
+ * @font: #hb_font_t from which to retrieve the value
+ * @constant: #hb_ot_math_constant_t representing an absolute height, kern,
+ * gap, drop, rise, shift, distance, etc
+ *
+ * Return value: the requested constant or 0
+ *
+ * Since: ????
+ **/
+hb_position_t
+hb_ot_layout_get_math_constant (hb_font_t *font,
+                                hb_ot_math_constant_t constant)
+{
+  const OT::MATH &math = _get_math (font->face);
+  if (math.has_math_constants()) {
+      hb_position_t value;
+      if (math.get_math_constants().get_value(font, constant, value))
+        return value;
+  }
+  return 0;
+}
+
+/**
+ * hb_ot_layout_get_math_percent:
```

Comment:
This does not mach the name of the method. But as I said before, I don’t think the granularity is really needed since the client will need to know any way which constants are percent and which are absolute values and returning `hb_position_t` for both shouldn’t harm (it is just a 32 bit integer).